src/Stratis.Features.FederatedPeg/Collateral/CollateralChecker.cs

```diff
@@ -211,7 +233,28 @@ public bool CheckCollateral(IFederationMember federationMember)
 
             lock (this.locker)
             {
-                return (this.depositsByAddress[member.CollateralMainchainAddress] ?? 0) >= member.CollateralAmount;
+                AddressIndexerData balanceData = this.balancesDataByAddress[member.CollateralMainchainAddress];
+
+                if (balanceData == null)
+                {
+                    // No data. Assume collateral is 0. It's ok if there is no collateral set for that fed member.
+                    this.logger.LogTrace("(-)[NO_DATA]");
+                    return 0 >= member.CollateralAmount;
```

Comment:
Why is this check being done? (Just trying to understand :)

Reply:
Just for readability. We use constant of 0 instead of actual balance when we can't get the balance. It's described above on ln240